src/coreclr/tools/aot/Mono.Linker.Tests/TestCasesRunner/AssemblyChecker.cs

```diff
@@ -155,19 +202,35 @@ void AddType (TypeDesc type)
 				}
 			}
 
-			bool AddMember (TypeSystemEntity entity)
+			void AddProperty (PropertyPseudoDesc property)
 			{
-				if (NameUtils.GetActualOriginDisplayName (entity) is string fullName &&
-					!linkedMembers.ContainsKey (fullName)) {
+				if (!ShouldIncludeProperty (property))
+					return;
 
-					linkedMembers.Add (fullName, entity);
-					return true;
-				}
+				if (!AddMember (property))
+					return;
 
-				return false;
+				if (property.OwningType is { } owningType)
+					AddType (owningType);
```

Comment:
AddProperty/AddEvent only gets called from AddMethod in which we already added the owning type. I think this will only be overhead by inserting the same owning type again... or is there something I'm missing

Reply:
Yes - I just wanted to make this "consistent". In a way the fact that we recognize properties only through the accessors is a weird part. But I can change this.

This will probably have to change a lot if we ever start differentiating between "Kept for runtime" and "Kept for reflection".